lib/include/openturns/OTtestcode.hxx

```diff
@@ -379,6 +379,15 @@ inline void assert_almost_equal(const SymmetricMatrix &a, const SymmetricMatrix
   }
 }
 
+template <typename T>
+void assert_equal(const T & a, const T & b, const String errMsg = "")
+{
+  if (a != b)
+  {
+    throw TestFailed(OSS() << "Value " << a << " is not equal to " << b << " " << errMsg);
+  }
+}
+
```

Comment:
seems a rebase is needed

Reply:
I would rather do this at the very end of the review process, because all review comments would be lost. Please ask again for this when ready for merging.

Reply:
I squashed commits and rebased the branch. This is OK for merging in my opinion.